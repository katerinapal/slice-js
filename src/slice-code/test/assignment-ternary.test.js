import {runAllCombosTests} from './helpers/utils'

// fit(
//   'orAndOr(false, false, true, true)',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/logical-expressions'), ({orAndOr}) => {
//     return [orAndOr(false, false, true, true)]
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/assignment-ternary'),
  methods: [
    {
      methodName: 'assignmentTernary',
      possibleArguments: [
        ['a'], ['b'],
      ],
    },
    {
      methodName: 'diacriticsClean',
      possibleArguments: [
        ['Some apple'], ['Some Google'],
      ],
    },
  ],
})
