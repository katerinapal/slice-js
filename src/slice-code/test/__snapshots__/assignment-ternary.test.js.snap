// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignmentTernary("a") && assignmentTernary("b") 1`] = `
"export { assignmentTernary };

const output = {
  map: {
    a: 'A',
  }
}

function assignmentTernary(letter) {
  let string = 'init'
  string += letter in output.map ? output.map[letter] : letter
  return string
}"
`;

exports[`assignmentTernary("a") 1`] = `
"export { assignmentTernary };

const output = {
  map: {
    a: 'A',
  }
}

function assignmentTernary(letter) {
  let string = 'init'
  string += output.map[letter]
  return string
}"
`;

exports[`assignmentTernary("b") 1`] = `
"export { assignmentTernary };

function assignmentTernary(letter) {
  let string = 'init'
  string += letter
  return string
}"
`;

exports[`diacriticsClean("Some Google") 1`] = `
"export { diacriticsClean };

function diacriticsClean(input) {
  var string = ''
  var letters = input.split('')
  var index = 0
  var length = letters.length
  var letter

  for (; index < length; index++) {
    letter = letters[index]
    string += letter
  }

  return string
}"
`;

exports[`diacriticsClean("Some apple") && diacriticsClean("Some Google") 1`] = `
"export { diacriticsClean };

const output = {
  map: {
    a: 'A',
  }
}

function diacriticsClean(input) {
  var string = ''
  var letters = input.split('')
  var index = 0
  var length = letters.length
  var letter

  for (; index < length; index++) {
    letter = letters[index]
    string += letter in output.map ? output.map[letter] : letter
  }

  return string
}"
`;

exports[`diacriticsClean("Some apple") 1`] = `
"export { diacriticsClean };

const output = {
  map: {
    a: 'A',
  }
}

function diacriticsClean(input) {
  var string = ''
  var letters = input.split('')
  var index = 0
  var length = letters.length
  var letter

  for (; index < length; index++) {
    letter = letters[index]
    string += letter in output.map ? output.map[letter] : letter
  }

  return string
}"
`;
